storage: replica GC check in TestReplicateRestartAfterTruncationWithRemoveAndReAdd is disabled

The following test appears to have failed:

[#21543](https://circleci.com/gh/cockroachdb/cockroach/21543):

```
I160818 20:10:41.638990 rpc/context.go:158  closing 127.0.0.1:58684
I160818 20:10:41.640477 http2_server.go:276  transport: http2Server.HandleStreams failed to read frame: read tcp 127.0.0.1:58684->127.0.0.1:59170: use of closed network connection
I160818 20:10:41.642668 storage/engine/rocksdb.go:402  closing in-memory rocksdb instance
I160818 20:10:41.643088 storage/engine/rocksdb.go:402  closing in-memory rocksdb instance
I160818 20:10:41.643645 storage/engine/rocksdb.go:402  closing in-memory rocksdb instance
--- FAIL: TestReplicateRestartAfterTruncationWithRemoveAndReAdd (45.37s)
    <autogenerated>:12: storage/client_raft_test.go:1180, condition failed to evaluate within 45s: expected replica to be garbage collected
=== RUN   TestReplicateRestartAfterTruncation
I160818 20:10:41.648450 storage/engine/rocksdb.go:354  opening in memory rocksdb instance
W160818 20:10:41.650633 gossip/gossip.go:1039  not connected to cluster; use --join to specify a connected node
I160818 20:10:41.655292 rpc/context.go:192  dialing 127.0.0.1:60668
I160818 20:10:41.660453 storage/store.go:968  store=1:1: failed initial metrics computation: store=1:1: system config not yet available
I160818 20:10:41.662858 storage/replica.go:2675  gossiping cluster id {"00000000-0000-0000-0000-000000000000"} from store 1, range 1
I160818 20:10:41.666659 storage/replica.go:2065  store=1:1 range=1 [/Min-/Max): applying command with forced error: storage/replica.go:1990: no-op on empty Raft entry
E160818 20:10:41.667495 storage/replica.go:2100  store=1:1 range=1 [/Min-/Max): error executing raft command: storage/replica.go:1990: no-op on empty Raft entry
I160818 20:10:41.668359 storage/replica_trigger.go:308  store=1:1 range=1 [/Min-/Max): new range lease replica {1 1 1} 1970-01-01 00:00:00 +0000 UTC 900ms following replica {0 0 0} 1970-01-01 00:00:00 +0000 UTC 0s [physicalTime=1970-01-01 00:00:00 +0000 UTC]
--
I160818 20:11:27.802632 stopper.go:408  quiesceing; tasks left:
36     gossip/infostore.go:290
I160818 20:11:27.813925 stopper.go:408  quiesceing; tasks left:
1      gossip/infostore.go:290
--- PASS: Example_rebalancing (1.02s)
FAIL
FAIL    github.com/cockroachdb/cockroach/storage    116.808s
=== RUN   TestBatchBasics
I160818 20:11:01.568540 storage/engine/rocksdb.go:354  opening in memory rocksdb instance
I160818 20:11:01.570555 storage/engine/rocksdb.go:402  closing in-memory rocksdb instance
--- PASS: TestBatchBasics (0.01s)
=== RUN   TestBatchRepr
I160818 20:11:01.584043 storage/engine/rocksdb.go:354  opening in memory rocksdb instance
I160818 20:11:01.585540 storage/engine/rocksdb.go:402  closing in-memory rocksdb instance
--- PASS: TestBatchRepr (0.01s)
=== RUN   TestBatchGet
I160818 20:11:01.592077 storage/engine/rocksdb.go:354  opening in memory rocksdb instance

```

Please assign, take a look and update the issue accordingly.

--- a/pkg/storage/client_raft_test.go
+++ b/pkg/storage/client_raft_test.go
@@ -1373,10 +1373,6 @@ func runReplicateRestartAfterTruncation(t *testing.T, removeBeforeTruncateAndReA
 		mtc.manualClock.Increment(int64(storage.ReplicaGCQueueInactivityThreshold + 1))
 		mtc.stores[1].ForceReplicaGCScanAndProcess()
 		util.SucceedsSoon(t, func() error {
-			if true {
-				// TODO(spencerkimball): fix the flakiness seen in #8670 and remove.
-				return nil
-			}
 			_, err := mtc.stores[1].GetReplica(rangeID)
 			if _, ok := err.(*roachpb.RangeNotFoundError); !ok {
 				return errors.Errorf("expected replica to be garbage collected")

--- a/pkg/storage/replica_gc_queue.go
+++ b/pkg/storage/replica_gc_queue.go
@@ -106,12 +106,9 @@ func (q *replicaGCQueue) shouldQueue(
 
 	lastActivity := hlc.ZeroTimestamp.Add(repl.store.startedAt, 0)
 
-	lease, nextLease := repl.getLease()
-	if lease != nil {
-		lastActivity.Forward(lease.Expiration)
-	}
-	if nextLease != nil {
-		lastActivity.Forward(nextLease.Expiration)
+	lease, _ := repl.getLease()
+	if lease != nil && lease.ProposedTS != nil {
+		lastActivity.Forward(*lease.ProposedTS)
 	}
 
 	var isCandidate bool
@@ -134,7 +131,7 @@ func replicaGCShouldQueueImpl(
 		priority = replicaGCPriorityCandidate
 	} else if now.Less(lastCheck.Add(ReplicaGCQueueInactivityThreshold.Nanoseconds(), 0)) {
 		// Return false immediately if the previous check was less than the
-		// check interval in the past. Note that we don't do this is the
+		// check interval in the past. Note that we don't do this if the
 		// replica is in candidate state, in which case we want to be more
 		// aggressive - a failed rebalance attempt could have checked this
 		// range, and candidate state suggests that a retry succeeded. See

